src/Features/Core/Portable/AddParameter/AbstractAddParameterCodeFixProvider.cs

```diff
@@ -71,15 +78,30 @@ internal abstract class AbstractAddParameterCodeFixProvider<
                 return null;
             }
 
+            if (this.CannotConvertDiagnosticIds.Contains(diagnostic.Id))
+            {
+                return null;
+            }
+
             return initialNode.GetAncestorsOrThis<TArgumentSyntax>()
                               .LastOrDefault(a => a.AncestorsAndSelf().Contains(node));
         }
 
-        private Task HandleInvocationExpressionAsync(
+        private async Task HandleInvocationExpressionAsync(
             CodeFixContext context, TInvocationExpressionSyntax invocationExpression, TArgumentSyntax argumentOpt)
         {
-            // Currently we only support this for 'new obj' calls.
-            return SpecializedTasks.EmptyTask;
+            var document = context.Document;
```

Comment:
@jinujoseph @mavasani at some point, will we go through all analyzer/fixer to change these to IOperation? when should we start ask people to stop using syntax node and use IOperation?

Reply:
if there is perf concern on IOperation, we should improve IOperation. we can only improve as we use it more. and I don't think we should stop it due to speculation. we have RPS, if RPS is green, it should be green. if we don't believe that is enough, we should add test that set the bar.

Reply:
I agree.  Can you work with @sharwell to get us to a good place there.  If/when IOperatoin is good enough, we can considre it for features like this.

However, as i mentioned above> Also, remember from the for-loop work, not all analyzer/fix work is necessarily amenable or better with IOperation.

The patterns and practices around writing these features are well understood with our existing APIs.  Sometimes they get a little better with IOp, sometimes they get worse (for example, you often have to know what 'odd' stuff IOp adds that you have to work around).  

In this case, this was built off hte work we already did for 'add parameter', so it makes sense to follow the same design as we already have there.